theano/gpuarray/opt.py

```diff
@@ -570,22 +833,29 @@ def local_gpu_pdbbreakpoint_op(node):
 
 @register_opt('fast_compile')
 @op_lifter([IfElse])
-def local_gpua_lazy_ifelse(node, context_name):
-    if node.op.gpu:
+@register_opt2([IfElse], 'fast_compile')
+def local_gpua_lazy_ifelse(op, context_name, inputs, outputs):
+    if op.gpu:
+        return
+    # this node is already on GPU, so don't change the graph
+    if isinstance(inputs[0].type, GpuArrayType):
```

Comment:
yes

On Thu, Jun 16, 2016 at 10:33 AM, Ramana Subramanyam <
notifications@github.com> wrote:

> In theano/gpuarray/opt.py
> <https://github.com/Theano/Theano/pull/4570#discussion_r67355623>:
>
> > @@ -570,22 +833,29 @@ def local_gpu_pdbbreakpoint_op(node):
> >
> >  @register_opt('fast_compile')
> >  @op_lifter([IfElse])
> > -def local_gpua_lazy_ifelse(node, context_name):
> > -    if node.op.gpu:
> > +@register_opt2([IfElse], 'fast_compile')
> > +def local_gpua_lazy_ifelse(op, context_name, inputs, outputs):
> > +    if op.gpu:
> > +        return
> > +    # this node is already on GPU, so don't change the graph
> > +    if isinstance(inputs[0].type, GpuArrayType):
>
> So, we're going to be aggressive ?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/Theano/Theano/pull/4570/files/e34b081d31be372573948d696154fb5dcd9e03ce#r67355623>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AALC-5_NaHI_iaVaYzYanXfJ64K6BiwDks5qMV66gaJpZM4Ir4rI>
> .
>